### ISSUE TITLE
terraform-provider-kubernetes_v2.11.0_x5 plugin crashed when add kubernetes_service_account

### ISSUE THREAD

**Issue by sjhuang93** [hashicorp/terraform-provider-kubernetes#1754]
### Terraform Version, Provider Version and Kubernetes Version

```
Terraform v1.2.3 on darwin_amd64
+ provider registry.terraform.io/aliyun/alicloud v1.173.0
+ provider registry.terraform.io/hashicorp/alicloud v1.173.0
+ provider registry.terraform.io/hashicorp/aws v4.18.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.11.0
```

### Affected Resource(s)
kubernetes_service_account

### Debug Output
```
module.k8s_pod_xxx.kubernetes_service_account.k8s_pod_service: Creating...
╷
│ Error: Request cancelled
│ 
│   with module.k8s_pod_xxx.kubernetes_service_account.k8s_pod_service,
│   on ../modules/eks_service/irsa.tf line 40, in resource "kubernetes_service_account" "k8s_pod_service":
│   40: resource "kubernetes_service_account" "k8s_pod_service" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-kubernetes_v2.11.0_x5 plugin:
```

### Panic Output

```
panic: set item just set doesn't exist

goroutine 112 [running]:
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).setSet(0xc0000c25d0, {0xc0010c7ee0, 0x1, 0x1}, {0x260cc00, 0xc00012b518}, 0xc0002b9180)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/field_writer_map.go:327 +0x992
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).set(0xc0000c25d0, {0xc0010c7ee0, 0x1, 0x1}, {0x260cc00, 0xc00012b518})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/field_writer_map.go:107 +0x14c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*MapFieldWriter).WriteField(0xc0000c25d0, {0xc0010c7ee0, 0x1, 0x1}, {0x260cc00, 0xc00012b518})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/field_writer_map.go:89 +0x3f9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*ResourceData).Set(0xc0000cbb00, {0x29930a6, 0x11}, {0x260cc00, 0xc00012b518})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/resource_data.go:227 +0x210
github.com/hashicorp/terraform-provider-kubernetes/kubernetes.resourceKubernetesServiceAccountRead({0x2c7ec48, 0xc001275380}, 0xc0000cbb00, {0x28acc80?, 0xc0000d0f40})
        github.com/hashicorp/terraform-provider-kubernetes/kubernetes/resource_kubernetes_service_account.go:281 +0x80c
github.com/hashicorp/terraform-provider-kubernetes/kubernetes.resourceKubernetesServiceAccountCreate({0x2c7ec48, 0xc001275380}, 0x270b620?, {0x28acc80?, 0xc0000d0f40})
        github.com/hashicorp/terraform-provider-kubernetes/kubernetes/resource_kubernetes_service_account.go:109 +0x8ab
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00027cb60, {0x2c7ec80, 0xc00089a870}, 0xd?, {0x28acc80, 0xc0000d0f40})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/resource.go:342 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00027cb60, {0x2c7ec80, 0xc00089a870}, 0xc0000bd1e0, 0xc0000cb980, {0x28acc80, 0xc0000d0f40})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/resource.go:472 +0x83a
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000d548, {0x2c7ec80?, 0xc00089a6f0?}, 0xc00135e8c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.12.0/helper/schema/grpc_provider.go:1021 +0xdaa
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.muxServer.ApplyResourceChange({0xc000e96b40, 0xc000e96ba0, {0xc000c96920, 0x2, 0x2}, 0xc000e96b70, 0xc000e53050, 0xc000c6ed00, 0xc000e96bd0}, {0x2c7ebd8, ...}, ...)
        github.com/hashicorp/terraform-plugin-mux@v0.6.0/tf5muxserver/mux_server_ApplyResourceChange.go:27 +0x142
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000587040, {0x2c7ec80?, 0xc000691230?}, 0xc0018d2230)
        github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/tf5server/server.go:812 +0x515
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x28b6580?, 0xc000587040}, {0x2c7ec80, 0xc000691230}, 0xc001274780, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.8.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000323500, {0x2c82cf8, 0xc0008444e0}, 0xc0018f6000, 0xc000ced1d0, 0x3a70160, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1282 +0xccf
google.golang.org/grpc.(*Server).handleStream(0xc000323500, {0x2c82cf8, 0xc0008444e0}, 0xc0018f6000, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1619 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.45.0/server.go:921 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.45.0/server.go:919 +0x28a

Error: The terraform-provider-kubernetes_v2.11.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Steps to Reproduce
1. `terraform init`
2. `terraform plan`
3. `terraform apply`

### Expected Behavior
Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

### Actual Behavior
`terraform apply`
Error: Request cancelled

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by arybolovlev** [comment]
Hi @sjhuang93,

Could you please share the affected resource code and let us know the Kubernetes version you are using?

Thank you.

**Comment by arybolovlev** [comment]
Thank you, @sjhuang93. It looks like a duplicate of [this](https://github.com/hashicorp/terraform-provider-kubernetes/issues/1741) issue. The crash happens when `image_pull_secret.name` is `null`. This issue has been addressed and fix will be available in `v2.12.0`.
